scripts/build_dependencies/build_step_tar.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+ const process = require("process");
+ const build_step_axios = require("./build_step_axios");
+ const utils = require('./build_utils');
+
+module.exports = {
+
+    tar : null,
+
+    /**
+     * Will download the file at the given url with the given version to the given path using tar.
+     * Will automatically either use or download the runtime-package for tar as needed.
+     */
+    performStep: async function (url, version, path) {
+        if (utils.npmCheckIfPackageExists("tar")) {
+            await this.fetchNativeCode(url, version, path);
+        } else {
+            await this.getPackageAndFetchNativeCode(url, version, path);
+        }
+    },
+
+    /**
+     * Will download the file at the given url with the given version to the given path using tar.
+     * Will ALWAYS download tar to the node_modules in scripts/build_dependencies/node_modules.
+     */
+    getPackageAndFetchNativeCode : async function (url, version, path) {
+        const workDir = path.join(__dirname, "../../")
+
+        process.chdir(__dirname);
+        if (this.tar == null) {
+            try {
+                utils.npmDownloadAndInstallRuntimePackage("tar", tar_version);
```

Comment:
there is no `tar_version` as well. So, again, make sure it works at least.

Reply:
Good catch, I missed that.

I did test it and it works though - it will just pull the latest version of the package if you do not manually specify a version. It just doesn't specify to NPM in the `npm install` call under the hood which version you want if you pass `null` or `undefined`.

Reply:
Yeah, unfortunately. Typescript will, but not Javascript. It's part of what makes it *really* hard to debug at times, from past experience.

Reply:
Okay - I did a bunch more tests and this was a good catch. I'm super glad you asked me to check because I probably would have missed it if I was not taking another careful look!

It was sort of working, but it was due to a bunch of other errors that basically made it not fully go down the code path! There was also some other cases there the version wasn't being used and the tar extract was pointing to the wrong place as well... So it worked, because I was running it in the CRT and already had the source code downloaded and in the right location, but if it was outside of that environment, it wouldn't have worked.

So... All of that is fixed now! I fully tested it and it all works and I made sure there was not a single error in the NPM build log. I missed it was doing some wonky things. Sigh - Javascript is *fun*.